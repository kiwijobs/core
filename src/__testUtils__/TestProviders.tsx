import React, { FC, Suspense } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const TestProviders: FC<{}> = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <Suspense fallback="mocked loading...">{children}</Suspense>
    </ThemeProvider>
  );
};

export const wrapper = TestProviders; // React Testing Library friendly name.
