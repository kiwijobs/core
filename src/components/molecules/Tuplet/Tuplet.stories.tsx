import React, { useState } from 'react';
import { storiesOf } from '@storybook/react';
import { Paper, Box, Logotype, Avatar } from '../..';
import { Tuplet } from './Tuplet';

storiesOf('Molecules/Tuplet', module)
  .add('Default', () => {
    const [active, setActive] = useState(0);

    return (
      <Paper m={4}>
        <Tuplet
          active={active}
          onChange={index => setActive(index)}
          buttons={['Dodaj z listy', 'ZaproÅ› nowych']}
        />
      </Paper>
    );
  })
  .add('Custom Nodes', () => {
    const [active, setActive] = useState(0);

    return (
      <Paper m={4}>
        <Tuplet
          active={active}
          onChange={index => setActive(index)}
          sx={{
            height: '60px',
          }}
          buttons={[
            <Box display="inline-flex" alignItems="center">
              <Logotype src="https://placebear.com/200/200" />
              <Box ml={2}>Companies</Box>
            </Box>,
            <Box display="inline-flex" alignItems="center">
              <Avatar src="https://placebear.com/200/200" />
              <Box ml={2}>Users</Box>
            </Box>,
          ]}
        />
      </Paper>
    );
  });
