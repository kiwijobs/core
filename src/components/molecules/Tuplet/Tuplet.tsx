import React, { FC, ReactNode, memo } from 'react';
import { Flex, BoxProps } from '../../quarks';
import { Button } from '../../atoms';

export const Tuplet: FC<ITupletProps> = memo(({ buttons, active = 0, onChange, sx, ...props }) => (
  <Flex
    sx={{
      width: '100%',
      alignItems: 'stretch',
      minHeight: '44px',
      ...sx,
    }}
    {...props}
  >
    {buttons.map((button, index) => (
      <Button
        key={index}
        width={1 / buttons.length}
        sx={{
          border: 1,
          color: 2,
          px: 2,
          display: 'flex',
          alignItems: 'center',
          justifyContent: 'center',

          '&:first-of-type': {
            borderTopLeftRadius: 1,
            borderBottomLeftRadius: 1,
          },
          '&:last-of-type': {
            borderTopRightRadius: 1,
            borderBottomRightRadius: 1,
          },

          ...(active === index && {
            borderWidth: '2px',
            borderColor: 'secondary',
            color: 1,
            pointerEvents: 'none',
          }),
        }}
        onClick={active === index ? undefined : event => onChange(index, event)}
      >
        {button}
      </Button>
    ))}
  </Flex>
));

interface ITupletProps extends Omit<BoxProps, 'onChange'> {
  buttons: ReactNode[];
  active: number;
  onChange(index: number, event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
}
