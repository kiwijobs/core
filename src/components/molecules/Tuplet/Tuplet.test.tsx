import React from 'react';
import { wrapper } from '../../../__testUtils__';
import { render, fireEvent } from '@testing-library/react';
import { Tuplet } from './Tuplet';

describe('Tuplet', () => {
  it('Matches snapshot', () => {
    expect(
      render(<Tuplet active={0} onChange={() => {}} buttons={['Left', 'Right']} />, {
        wrapper,
      }).container
    ).toMatchSnapshot();
  });

  it('"onChange" has correct params', () => {
    const mockOnChange = jest.fn();

    const { getByText } = render(
      <Tuplet active={0} onChange={mockOnChange} buttons={['Left', 'Right']} />,
      {
        wrapper,
      }
    );

    fireEvent.click(getByText('Left'));
    expect(mockOnChange).not.toHaveBeenCalled();

    fireEvent.click(getByText('Right'));
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
