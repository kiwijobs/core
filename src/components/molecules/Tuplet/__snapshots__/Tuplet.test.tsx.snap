// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tuplet Matches snapshot 1`] = `
.c1 {
  cursor: pointer;
  border: 1px solid rgb(228,230,237);
  color: rgb(39,50,55);
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-width: 2px;
  border-color: rgb(83,109,254);
  pointer-events: none;
  width: 50%;
}

.c1:first-of-type {
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
}

.c1:last-of-type {
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
}

.c2 {
  cursor: pointer;
  border: 1px solid rgb(228,230,237);
  color: rgb(133,136,149);
  padding-left: 8px;
  padding-right: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50%;
}

.c2:first-of-type {
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
}

.c2:last-of-type {
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
}

.c0 {
  width: 100%;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  min-height: 44px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      type="button"
      width="0.5"
    >
      Left
    </button>
    <button
      class="c2"
      type="button"
      width="0.5"
    >
      Right
    </button>
  </div>
</div>
`;
